Python parser

guihomework opened the pull request:
Following the discussions in https://github.com/ros-planning/srdfdom/issues/8 this PR adds support for a python parser in srdfdom based on the work done in urdf_parser_py and on its reflection library

guihomework committed:
Added a python parser based on urdf_parser_py and using its reflection interface

guihomework committed:
Fixed missing install

davetcoleman commented on package.xml:
```diff
@@ -1,11 +1,16 @@
   <author email="isucan@willowgarage.edu">Ioan Sucan</author>
+  <author email="gwalck@techfak.uni-bielefeld.de">Guillaume Walck</author>
+  <author>Wim Meeussen</author>
+  <author>John Hsu</author>
```
Why are these guys being added as authors? They have never been involved with SRDF afaik

guihomework replied:
It comes from the urdfdom from which I copied the parser and adapted it to srdf (https://raw.githubusercontent.com/ros-gbp/urdfdom-release/debian/indigo/trusty/urdfdom/package.xml) 
But if looking in details, the people added in the README.md are the correct ones I derived the work from. 
I don't know how to acknowledge these people.

davetcoleman replied:
I think what you have in the README is sufficient, if we made everyone authors of new packages based on references to old code, we would have very long author lists. Generally, only major contributors to a package who have specifically worked on that one claim authorship. I think you should remove the other ones.

davetcoleman commented:
This is an impressive PR - thanks for creating tests, a demo script, improving documentation, etc. I'm not really a python programmer so can't speak to that part's quality, but from my review everything looks good, except:
- Could you document the display_srdf script in the README?
- Could you document how to use/run the test? This might be obvious and a standard ROS thing, but I always forget

guihomework commented:
while adding more tests (the same as in the cpp parser) I found some tiny bugs in the python parser. Currently fixing them.
Why is travis building for precise ? This is indigo-devel branch and I don't remember indigo was supported on precise.

davetcoleman commented:
travis does not support anything newer, unfortunately. we've turned off travis support in most of the moveit repos already, actually. i just turned it off for this one, too.

guihomework committed:
Fixed group_state parsing and changed chain as an aggregate
Renamed groups as subgroups when integrated in a group
Added the cpp tests in the python test

guihomework committed:
Fixed authors, added doc

guihomework committed:
Fixed path to resource in python test to work for rostest
Added how to run test

guihomework committed:
removed unwanted python compiled file

guihomework commented:
I did the suggested changes. I also noticed the cpp_parser test is not started with rostest but I will create a new PR for that after this one is merged (to separate issues)

davetcoleman commented:
Great, thanks for addressing my feedback!

ugocupcic commented:
@davetcoleman Could you tell me if you're planning to do a new release in the near future? We'd like to use @guihomework new feature :).

Cheers,

davetcoleman commented:
Just released, again sorry for delay

ugocupcic commented:
@davetcoleman thanks !
